crates/extensions/c8y_mapper_ext/src/operations/handlers/custom_operation.rs

```diff
@@ -211,6 +240,73 @@ mod tests {
         assert_received_contains_str(&mut mqtt, [("c8y/s/us", "506,1234")]).await;
     }
 
+    #[tokio::test]
+    async fn handle_custom_operation_successful_cmd_for_main_device_with_result() {
+        let ttd = TempTedgeDir::new();
+        let config = C8yMapperConfig {
+            smartrest_use_operation_id: true,
+            ..test_mapper_config(&ttd)
+        };
+        let test_handle = spawn_c8y_mapper_actor_with_config(&ttd, config, true).await;
+        let TestHandle { mqtt, .. } = test_handle;
+        let mut mqtt = mqtt.with_timeout(TEST_TIMEOUT_MS);
+
+        skip_init_messages(&mut mqtt).await;
+
+        // Simulate custom operation command with "successful" state
+        mqtt.send(MqttMessage::new(
+            &Topic::new_unchecked("te/device/main///cmd/command/c8y-mapper-1234"),
+            json!({
+                "status": "successful",
+                "text": "do something",
+                "result": ["on","off","on"]
+            })
+            .to_string(),
+        ))
+        .await
+        .expect("Send failed");
+
+        // Expect `506` smartrest message on `c8y/s/us`.
+        assert_received_contains_str(&mut mqtt, [("c8y/s/us", "506,1234,on,off,on")]).await;
+    }
+
+    #[tokio::test]
+    async fn handle_custom_operation_successful_cmd_with_unsupported_result() {
+        let ttd = TempTedgeDir::new();
+        let config = C8yMapperConfig {
+            smartrest_use_operation_id: false,
+            ..test_mapper_config(&ttd)
+        };
+        let test_handle = spawn_c8y_mapper_actor_with_config(&ttd, config, true).await;
+        let TestHandle { mqtt, .. } = test_handle;
+        let mut mqtt = mqtt.with_timeout(TEST_TIMEOUT_MS);
+
+        skip_init_messages(&mut mqtt).await;
+
+        // Simulate custom operation command with "successful" state
+        mqtt.send(MqttMessage::new(
+            &Topic::new_unchecked("te/device/main///cmd/command/c8y-mapper-1234"),
+            json!({
+                "status": "successful",
+                "text": "do something",
+                "result": true
+            })
+            .to_string(),
+        ))
+        .await
+        .expect("Send failed");
+
+        // Expect `502` smartrest message on `c8y/s/us` as 'result' contains unsupported type
+        assert_received_contains_str(
+            &mut mqtt,
+            [(
+                "c8y/s/us",
+                "502,command,'result' field must have String or Array",
```

Comment:
`command` is the name from the thin-edge MQTT message. It should be the Cumulocity operation name, which should be available from a custom operation handler definition...

Reply:
Better to fix this bug now as it defeats the purpose of this PR.

Reply:
It's addressed in this PR.